### PR TITLE
admin: Add support for comment search by URL

### PR DESCRIPTION
**Pull request by pkvach**
## Checklist
- [x] All new and existing **tests are passing**
- [x] I have added an entry to `CHANGES.rst` because this is a user-facing change or an important bugfix
- [x] I have written **proper commit message(s)**
      

## What changes does this Pull Request introduce?

Added support for searching comments by URL in the admin interface. This includes changes to the backend to handle comment URL queries and extract comment IDs, and frontend changes to add a search input to the admin page.

- Updated admin CSS to include styles for the new search input and button.

- Modified the fetchall method in comments.py to accept an optional comment_id parameter for fetching a specific comment by ID.

- Updated the admin HTML template to include a search form that allows inputting a comment URL for search.

- Updated the admin method in views/comments.py to handle the comment_url query parameter, parse the URL to extract the comment ID, and fetch the corresponding comment.

## Why is this necessary?

Closes https://github.com/isso-comments/isso/issues/642

### Demo:
image

image

### CODE REVIEW

File: isso/views/comments.py [hunk 1]
```diff
@@ -1396,10 +1398,25 @@ def admin(self, env, req):
...
+        # Search for a specific comment by URL
+        if comment_url:
+            try:
+                # Parse the comment URL to extract the comment ID from the fragment
+                parsed_url = urlparse(comment_url)
+                fragment = parsed_url.fragment
+                comment_id = int(fragment.split('-')[-1])
+
+                comments = self.comments.fetchall(comment_id=comment_id, limit=1)
+            except Exception:
```
Comment (jelmer) [inline isso/views/comments.py:1412]: Please catch more specific exceptions and more specific to what's raising them; "Exception" is too broad and can catch exceptions we don't intend to.

**Review by jelmer — APPROVED** [review]

**Comment by ix5** [comment]
If I see this correctly, the search only succeeds if you supply both the **thread** URL as well as a per-thread **comment** id.

What about admins who would want to look up only `http://localhost:8080/demo/`, e.g. the **thread** URL, to moderate all comments for a particular blogpost? Since this was merged already, could you add that in a follow-up PR?

**Comment by ix5** [comment]
> Yes, this PR implements the requested search for specific comments. I agree that we could make it even better by allowing comments to be searchable by thread URL.

Could you create a follow-up PR? I would argue that admins most often would search for threads, not comments.

**Comment by pkvach** [comment]
Added a new PR to search for comments by thread URL: https://github.com/isso-comments/isso/pull/1020

**Pull request merged**